packages/x-components/src/x-modules/history-queries/components/__tests__/my-history.spec.ts

```diff
@@ -61,13 +68,60 @@ describe('testing MyHistory component', () => {
     expect(wrapper.html()).toEqual('');
   });
 
-  it('renders the list of searched queries', async () => {
-    const { search, getListItems } = renderMyHistory();
+  it('renders the list of searched queries group by date', () => {
+    const historyQueriesGroupByDate: HistoryQuery[] = [
+      {
+        query: 'lego',
+        timestamp: 1650286901802,
+        modelName: 'HistoryQuery'
+      },
+      {
+        query: 'barbie',
+        timestamp: 1650286895254,
+        modelName: 'HistoryQuery'
+      },
+      {
+        query: 'truck',
+        timestamp: 1649230515242,
+        modelName: 'HistoryQuery'
+      },
+      {
+        query: 'doll',
+        timestamp: 1649230513535,
+        modelName: 'HistoryQuery'
+      }
+    ];
+
+    const expectedResult: Record<string, HistoryQuery[]> = {
+      'Monday, April 18, 2022': [historyQueriesGroupByDate[0], historyQueriesGroupByDate[1]],
+      'Wednesday, April 6, 2022': [historyQueriesGroupByDate[2], historyQueriesGroupByDate[3]]
+    };
 
-    await search('lego');
-    const suggestionsWrappers = getListItems();
-    expect(suggestionsWrappers.wrappers).toHaveLength(1);
-    expect(suggestionsWrappers.at(0).text()).toEqual('lego ✕');
+    const { findAllInWrapper } = renderMyHistory({
+      historyQueries: historyQueriesGroupByDate
+    });
+
+    const historyWrappers = findAllInWrapper('my-history-item');
+
+    historyWrappers.forEach((dateWrapper, index) => {
+      const date = dateWrapper.find(getDataTestSelector('my-history-date')).text();
+      expect(Object.keys(expectedResult)[index]).toBe(date);
+      dateWrapper
+        .findAll(getDataTestSelector('my-history-query'))
+        .wrappers.forEach((wrapper, queryIndex) => {
+          expect(wrapper.text()).toEqual(expectedResult[date][queryIndex].query);
+        });
+      dateWrapper
+        .findAll(getDataTestSelector('my-history-time'))
+        .wrappers.forEach((wrapper, queryIndex) => {
+          expect(wrapper.text()).toEqual(
+            new Date(expectedResult[date][queryIndex].timestamp).toLocaleTimeString([], {
+              hour: '2-digit',
+              minute: '2-digit'
+            })
```

Comment:
Should we extract the `formatTime` method to the utils folder? I suggested a change in that function that could make this test fail based on configured the `locale`.